pkg/k8shandler/secret.go

```diff
@@ -45,12 +46,16 @@ func (clusterRequest *ClusterLoggingRequest) CreateOrUpdateSecret(secret *core.S
 				}
 				return fmt.Errorf("Failed to get %v secret: %v", secret.Name, err)
 			}
-
-			current.Data = secret.Data
-			if err = clusterRequest.Update(current); err != nil {
-				return err
+			if reflect.DeepEqual(current.Data, secret.Data) {
+				// identical; no need to update.
+				return nil
+			} else {
```

Comment:
you can remove the else block around the code for readability since you return early when no update is required